src/test/ExtractionClient.test.ts

```diff
@@ -55,6 +55,36 @@ describe("ExtractionClient", () => {
       "https://api.bentley.com/insights/reporting/datasources/imodels/iModelId/extraction/run",
       "pass",
     )).to.be.true;
+    expect(requestStub.calledWith(
+      "POST",
+      "auth",
+      undefined,
+    )).to.be.true;
+  });
+
+  it("run extraction with parameters", async () => {
+    const returns = {
+      run: {
+        id: 1,
+      },
+    };
+    fetchStub.resolves(returns);
+    const extractionRequest: ExtractionRunRequest = {
+      changesetId: "changesetId",
+      mappings: [{ id: "mappingId" }],
+      ecInstanceIds: ["ecInstanceId"],
+    };
+    const extraction = await extractionClient.runExtraction("auth", "iModelId", extractionRequest);
+    expect(extraction.id).to.be.eq(1);
+    expect(fetchStub.calledWith(
+      "https://api.bentley.com/insights/reporting/datasources/imodels/iModelId/extraction/run",
+      "pass",
+    )).to.be.true;
+    expect(requestStub.calledWith(
+      "POST",
+      "auth",
+      JSON.stringify(extractionRequest),
```

Comment:
That's just a copy paste of the implementation. I would rather get the argument from a call and assert each of the objects properties individually.

Reply:
`createRequest` is called with just 3 strings.

Reply:
`JSON.parse(requestStub.getCall(...).args[2])` and it is no longer a string.

Reply:
I don't see point in doing this. No other test is parsing request bodies. And if we want to not rely on `JSON.stringify` we can just expect a preserialized string.